lib/travis/model/job.rb

```diff
@@ -122,13 +122,8 @@ def decrypted_config
     normalize_config(self.config).deep_dup.tap do |config|
       config[:env] = process_env(config[:env]) { |env| decrypt_env(env) } if config[:env]
       config[:global_env] = process_env(config[:global_env]) { |env| decrypt_env(env) } if config[:global_env]
-      if config[:addons]
-        if addons_enabled?
-          config[:addons] = decrypt_addons(config[:addons])
-        else
-          delete_addons(config)
-        end
-      end
+      delete_addons(config) if config[:addons] && !addons_enabled?
+      config[:addons] = decrypt_addons(config[:addons]) if config[:addons]
```

Comment:
@santiycr @sebv Could you rework this part of logical flow before I merge this? A more straightforward

```ruby
if config[:addons]
  if addons_enabled?
    config[:addons] = decrypt_addons(config[:addons])
  else
    delete_addons(config)
    config[:addons] = decrypt_addons(config[:addons])
  end
end
```

seems more readable to me.